src/Components/Task.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+
+const Task = ({ description, setTaskList, tasks, id, setTaskToBeEdited, setNewDescription }) => {
+    const [isStrikeThrough, setStrikeThrough] = useState(false);
+
+    const removeTask = (id) => {
+        return setTaskList(tasks.filter(task => task.id !== id))
+    };
+
+    useEffect(() => {
+        const modal = document.querySelector('.modal');
+        const overlay = document.querySelector('.overlay');
+        const editButtons = Array.from(document.querySelectorAll('.button__edit'));
+        
+        function showModal() {
+            modal.classList.remove('hidden');
+            overlay.classList.remove('hidden');
+        }
+
+        editButtons.forEach(editButton => editButton.addEventListener('click', showModal));
+    }, []);
+    
+    const showTaskEditor = (event) => {
+        event = event || window.event;
+        let src = event.target || event.srcElement;
+        const paragraph = src.closest('.task').firstChild.innerHTML;
+        tasks.forEach(task => {
+            if (task.description === paragraph) {
+                setTaskToBeEdited(task);
+                setNewDescription(task.description);
+            }
+        })
+        const modal = document.querySelector('.modal');
+        const overlay = document.querySelector('.overlay');
+        const editButtons = Array.from(document.querySelectorAll('.button__edit'));
+        
+        function showModal() {
+            modal.classList.remove('hidden');
+            overlay.classList.remove('hidden');
+        }
+
+        editButtons.forEach(editButton => editButton.addEventListener('click', showModal));
```

Comment:
Same comments for the code here also as mentioned above. If we declare this logic in the component we will not have to rewrite this code again and again.